opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/components.py

```diff
@@ -110,10 +120,30 @@ def import_exporters(
     return trace_exporters, metric_exporters
 
 
+def import_ids_generator(ids_generator_name: str) -> trace.IdsGenerator:
```

Comment:
nitpick: this whole thing looks refactorable to share the same logic as import_exporters, if you make things like the class parameterizable.

Reply:
I've added what I think is as much generalizable as makes sense. I think it may be more code than was worth it, but when we make more of the `TracerProvider` configurable in the future I can see how it will be worth it :)

Reply:
looks good! yeah the parsing through the logic is what I had it mind. You could even generalize the `isinstance` calls, but maybe that's overkill.